### PR TITLE
Transactions list

### PR DESCRIPTION
**Pull request by tmurawsk**
Added endpoint showing lists of incoming and outcoming transactions from given addres

### CODE REVIEW

File: backend/src/main/java/pl/edu/pw/elka/transactions/TransactionsController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    private final TransactionsFacade facade;
+
+    public TransactionsController(TransactionsFacade facade) {
+        this.facade = facade;
+    }
+
+    @GetMapping(value = "/ethereum/transactions/{address}")
+    public ResponseEntity<TransactionListsDto> fetchTransactionsLists(@PathVariable("address") String address) {
+        TransactionListsDto dto = facade.fetchTransactionLists(address);
+        return ResponseEntity.ok(dto);
```
Comment (mstar95) [inline backend/src/main/java/pl/edu/pw/elka/transactions/TransactionsController.java]: @RestController ma juz @ResponseBody, dla statusu 200 mozna zwrocic od razu TransactionListsDto

File: backend/src/main/java/pl/edu/pw/elka/transactions/TransactionsController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.pw.elka.transactions.domain.TransactionsFacade;
+import pl.edu.pw.elka.transactions.dto.TransactionListsDto;
+
+@RestController
+public class TransactionsController {
```
Comment (mstar95) [inline backend/src/main/java/pl/edu/pw/elka/transactions/TransactionsController.java]: Moze byc package-scoped

File: backend/src/main/java/pl/edu/pw/elka/transactions/domain/EtherscanTransactionsConnector.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        fillTransactionArrays(address, transactionsStringJson, incomeTransactionsArray, outcomeTransactionsArray);
+
+        return new TransactionLists(incomeTransactionsArray, outcomeTransactionsArray);
+    }
+
+    private void fillTransactionArrays(String address, String jsonData, ArrayList<TransactionDetails> incomeArray, ArrayList<TransactionDetails> outcomeArray) {
+        JSONObject givenJsonObject = new JSONObject(jsonData);
+        JSONArray givenJsonArray = givenJsonObject.getJSONArray("result");
+
+        for(int i = 0; i < givenJsonArray.length(); i++){
```
Comment (mstar95) [inline backend/src/main/java/pl/edu/pw/elka/transactions/domain/EtherscanTransactionsConnector.java]: Polecam stream

File: backend/src/main/java/pl/edu/pw/elka/transactions/domain/EtherscanTransactionsConnector.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            JSONObject object = givenJsonArray.getJSONObject(i);
+
+            if(object.getString("from").equals(address)) {
```
Comment (mstar95) [inline backend/src/main/java/pl/edu/pw/elka/transactions/domain/EtherscanTransactionsConnector.java]: rozbic na dwie metody

File: backend/src/main/java/pl/edu/pw/elka/transactions/domain/TransactionLists.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+@Getter
+@RequiredArgsConstructor
+@Builder
+class TransactionLists {
+
+    private ArrayList<TransactionDetails> incomeTransactionsList;
+    private ArrayList<TransactionDetails> outcomeTransactionsList;
+
+    public TransactionLists(ArrayList<TransactionDetails> arrayIn, ArrayList<TransactionDetails> arrayOut) {
```
Comment (mstar95) [inline backend/src/main/java/pl/edu/pw/elka/transactions/domain/TransactionLists.java]: PackageScoped klasa a metoda public XDDD, idea nic nie podpowiada, no chyba ze eclipse XD

**Comment by wieczorekm** [comment]
Done in #11

**Pull request closed without merging**